### ISSUE TITLE
How do you save ResNet-34 that is implemented from pages 478-479?

### ISSUE THREAD

**Issue by OrdinaryHacker101** [ageron/handson-ml2#450]
Hi Mr. Geron;

I am a teen fan of yours reading this book, and I have experimented with building a custom Residual Unit:
![Screenshot 2021-06-27 093424](https://user-images.githubusercontent.com/64097727/123546543-fdb49280-d72a-11eb-8ce7-13f6f904bc71.png)

However whenever I do model.save("test.h5") , I get this error:
![Screenshot 2021-06-27 093706](https://user-images.githubusercontent.com/64097727/123546608-4c622c80-d72b-11eb-957d-7e0b7cc8bde4.png)

P. S. Additional question: I am currently using this custom ResNet to do some behavioral cloning on the VizDoom game environment. My data, which consists of recordings of me playing, seems to be really noisy, and no matter how i train my model the lowest binary cross entropy loss is 0.659. Can you please give me some advice?

THANK YOU SOO MUCH
- Your Fan

**Comment by OrdinaryHacker101** [comment]
Wait I noticed that the loss of 0.659 is when I intentionally overfit my model on only 32 instances: if I train it on 10 entire videos the loss becomes 0.71. Sorry for the misunderstanding!

**Comment by ageron** [comment]
Hi @OrdinaryHacker101 ,
Thanks for your question (and for your kind words!).
The first issue might be a TensorFlow bug. Could you please provide a minimal code example that reproduces the bug? Also, perhaps try using a the file extension `.ckpt` instead of `.h5` and see if that solves the issue?
If I understand correctly, the second issue is solved, right?

**Comment by OrdinaryHacker101** [comment]
@ageron THANK YOU MR. GERON I solved both issues: I found out that it worked when I didn't save certain things e.g. skip layers, etc.
